src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ReadClassifier.java

```diff
@@ -0,0 +1,124 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import com.google.common.annotations.VisibleForTesting;
+import htsjdk.samtools.CigarElement;
+import htsjdk.samtools.CigarOperator;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.util.*;
+import java.util.function.Function;
+
+// TODO: figure out a way to include funky template size data without swamping ourselves in a sea of evidence
+// TODO: maybe count each such read as fractional evidence?
+
+/**
+ * Figures out what kind of BreakpointEvidence, if any, a read represents.
+ */
+public class ReadClassifier implements Function<GATKRead, Iterator<BreakpointEvidence>> {
+    @VisibleForTesting static final int MIN_SOFT_CLIP_LEN = 30; // minimum length of an interesting soft clip
```

Comment:
Could you make these into parameters of the command line tool, and add a ticket saying we should test different parameter values and weighting/clustering strategies?